test_scripts/RC/OnRCStatus/commonOnRCStatus.lua

```diff
@@ -0,0 +1,250 @@
+    struct.moduleType = "AUDIO"
+    struct.audioControlData = {
+      source = "CD",
+      keepContext = false,
+      volume = 50,
+      equalizerSettings = {
+        channelId = 10,
+        channelName = "Channel 1",
+        channelSetting = 50 }
+    }
+  elseif "LIGHT" == pModuleType then
+    struct.moduleType = "LIGHT"
+    struct.lightControlData = {
+      lightState = {
+        id = "FRONT_LEFT_HIGH_BEAM",
+        status = "ON",
+        density = 0.2,
+        sRGBColor = "red"
+      }
+    }
+  elseif "HMI_SETTINGS" == pModuleType then
+    struct.moduleType = "HMI_SETTINGS"
+    struct.hmiSettingsControlData = {
+      displayMode = "DAY",
+      temperatureUnit = "CELSIUS",
+      distanceUnit = "KILOMETERS"
+    }
+  end
+  return struct
+end
+
+function commonOnRCStatus.postconditions()
+  commonRC.postconditions()
+end
+
+function commonOnRCStatus.SetModuleStatus(pFreeMod, pAllocatedMod, pModule)
+  local ModulesStatus = { }
+  table.insert(pAllocatedMod, pModule)
+  for key, value in pairs(pFreeMod) do
+    if pModule == value then
+      table.remove(pFreeMod, key)
+    end
+  end
+  ModulesStatus.freeModules = pFreeMod
+  ModulesStatus.allocatedModules = pAllocatedMod
+  return ModulesStatus
+end
+
+function commonOnRCStatus.SetModuleStatusByDeallocation(pFreeMod, pAllocatedMod, pModule)
+  local ModulesStatus = { }
+  table.insert(pFreeMod, pModule)
+  for key, value in pairs(pAllocatedMod) do
+    if pModule == value then
+      table.remove(pAllocatedMod, key)
+    end
+  end
+  ModulesStatus.freeModules = pFreeMod
+  ModulesStatus.allocatedModules = pAllocatedMod
+  return ModulesStatus
+end
+
+function commonOnRCStatus.unregisterApp(pAppId)
+  if not pAppId then pAppId = 1 end
+  commonRC.unregisterApp(pAppId, test)
+end
+
+function commonOnRCStatus.defineRAMode(pAllowed, pAccessMode)
+  commonRC.defineRAMode(pAllowed, pAccessMode, test)
+end
+
+function commonOnRCStatus.ConsentRejected(pModuleType, pRPC, pAppId)
+  if not pAppId then pAppId = 1 end
+  local mobSession = commonOnRCStatus.getMobileSession(pAppId)
+  local cid = mobSession:SendRPC(commonRC.getAppEventName(pRPC), commonRC.getAppRequestParams(pRPC, pModuleType))
+  local consentRPC = "GetInteriorVehicleDataConsent"
+  EXPECT_HMICALL(commonRC.getHMIEventName(consentRPC), commonRC.getHMIRequestParams(consentRPC, pModuleType, pAppId))
+  :Do(function(_, data)
+      test.hmiConnection:SendError(data.id, data.method, "REJECTED", " Driver rejects request ")
+      EXPECT_HMICALL(commonRC.getHMIEventName(pRPC), commonRC.getHMIRequestParams(pRPC, pModuleType, pAppId))
+      :Times(0)
+    end)
+  mobSession:ExpectResponse(cid, { success = false, resultCode = "REJECTED", info = " Driver rejects request " })
+end
+
+function commonOnRCStatus.rpcAllowedWithConsent(pModuleType, pAppId, pRPC)
+  commonRC.rpcAllowedWithConsent(pModuleType, pAppId, pRPC, test)
+end
+
+function commonOnRCStatus.setVehicleData(pModuleType, pAppId)
```

Comment:
@GetmanetsIrina Please notice instead of this function ```rpcAllowed()``` one from ```commonRC``` module can be used

Reply:
Locally changed.